src/flows/pipes/RawFluxEditor/view.tsx

```diff
@@ -39,86 +46,50 @@ const FluxMonacoEditor = lazy(() =>
 )
 
 const Query: FC<PipeProp> = ({Context}) => {
-  const {id, data, update} = useContext(PipeContext)
-  const {hideSub, id: showId, show, showSub} = useContext(SidebarContext)
-  const [editorInstance, setEditorInstance] = useState<EditorType>(null)
+  const {id, data} = useContext(PipeContext)
+  const {hideSub, id: showId, show, showSub, register} = useContext(
+    SidebarContext
+  )
+  const editorContext = useContext(EditorContext)
+  const {register: setEditorInstance, inject, updateText} = editorContext
   const {queries, activeQuery} = data
   const query = queries[activeQuery]
 
-  const updateText = useCallback(
-    text => {
-      const _queries = queries.slice()
-      _queries[activeQuery] = {
-        ...queries[activeQuery],
-        text,
-      }
-
-      update({queries: _queries})
-    },
-    [update, queries, activeQuery]
-  )
+  useEffect(() => {
+    if (isFlagEnabled('fluxInjectSecrets')) {
+      register(id, [
+        {
+          title: 'RawFluxEditor actions',
+          actions: [
+            {
+              title: 'Inject Secret',
+              disable: () => false,
+              menu: <SecretsList context={editorContext} />,
```

Comment:
just pass the inject here instead of the entire context

Reply:
done